server/container_create.go

```diff
@@ -62,27 +61,17 @@ func (s *Server) CreateContainer(ctx context.Context, req *pb.CreateContainerReq
 		return nil, err
 	}
 
-	// containerDir is the dir for the container bundle.
-	containerDir := filepath.Join(s.runtime.ContainerDir(), containerID)
 	defer func() {
 		if err != nil {
 			s.releaseContainerName(containerName)
-			err1 := os.RemoveAll(containerDir)
+			err1 := s.storage.DeleteContainer(containerID)
```

Comment:
This should probably be setup in a defer after createSandboxContainer otherwise there is no guarantee whether storage was even setup for the container.